lib/puppet/provider/augeas/augeas.rb

```diff
@@ -203,7 +203,7 @@ def close_augeas
 
   def is_numeric?(s)
     case s
-    when Fixnum
+    when Integer,Float
```

Comment:
Take a look at the regular expression used for the String a couple of lines down. It accepts floats. Seems rather odd that the string form would, but the numeric form would not. One of them should be adjusted.